File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    var ans = [];
    var top_score = score.slice(0, k);
    for (let i = 0; i < score.length; i++) {
        if (i < k) ans.push(Math.min(...score.slice(0, i+1)));
        else {
            if (Math.min(...top_score) < score[i]) {
                top_score.splice(top_score.indexOf(Math.min(...top_score)), 1);
                top_score.push(score[i]);
            }
            ans.push(Math.min(...top_score));
        }
    }
    return ans;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
import heapq as hq

def solution(k, score):
    top_k = []  # min-heap
    ans = []
    for s in score:
        if len(top_k) >= k:
            if top_k[0] < s:
                hq.heappop(top_k)
                hq.heappush(top_k, s)
        
        else:  # len(top_k) < k
            hq.heappush(top_k, s)
            
        ans.append(top_k[0])
        
    return ans
        
    
            
            
    